bin/bench-prover/src/utils.rs

```diff
@@ -0,0 +1,86 @@
+use std::{
+    env,
+    fs::{self},
+    io::Write,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use serde::Deserialize;
+use serde_json::{Result as JsonResult, Value, json};
+
+pub fn cargo_target_directory() -> Option<PathBuf> {
+    #[derive(Deserialize)]
+    struct Metadata {
+        target_directory: PathBuf,
+    }
+
+    env::var_os("CARGO_TARGET_DIR").map(PathBuf::from).or_else(|| {
```

Comment:
Question: When would this environment variable not be set? Isn't this always set by cargo, so do we have to handle the `None` case?